Framework/Geometry/src/Rendering/RenderingHelpers.cpp

```diff
@@ -0,0 +1,438 @@
+        trian.Get(index1, index2, index3);
+        gp_Pnt point1 = tab.Value(index1);
+        gp_Pnt point2 = tab.Value(index2);
+        gp_Pnt point3 = tab.Value(index3);
+        gp_XYZ pt1 = tab.Value(index1).XYZ();
+        gp_XYZ pt2 = tab.Value(index2).XYZ();
+        gp_XYZ pt3 = tab.Value(index3).XYZ();
+
+        gp_XYZ v1 = pt2 - pt1;
+        gp_XYZ v2 = pt3 - pt2;
+
+        gp_XYZ normal = v1 ^ v2;
+        normal.Normalize();
+        glNormal3d(normal.X(), normal.Y(), normal.Z());
+        glVertex3d(point1.X(), point1.Y(), point1.Z());
+        glVertex3d(point2.X(), point2.Y(), point2.Z());
+        glVertex3d(point3.X(), point3.Y(), point3.Z());
+      }
+    }
+  }
+  glEnd();
+}
+#endif
+
+// Render Bitmap for RectangularDetector
+void render(const RectangularDetector &rectDet) {
+  // Because texture colours are combined with the geometry colour
+  // make sure the current colour is white
+  glColor3f(1.0f, 1.0f, 1.0f);
+
+  // Nearest-neighbor scaling
+  GLint texParam = GL_NEAREST;
+  glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MIN_FILTER, texParam);
+  glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MAG_FILTER, texParam);
+
+  glEnable(GL_TEXTURE_2D); // enable texture mapping
+
+  int texx, texy;
+  rectDet.getTextureSize(texx, texy);
+  double tex_frac_x = (1.0 * rectDet.xpixels()) / (texx);
+  double tex_frac_y = (1.0 * rectDet.ypixels()) / (texy);
+
+  glBegin(GL_QUADS);
+
+  glTexCoord2f(0.0, 0.0);
+  V3D pos;
+  pos = rectDet.getRelativePosAtXY(0, 0);
+  pos += V3D(rectDet.xstep() * (-0.5), rectDet.ystep() * (-0.5),
+             0.0); // Adjust to account for the size of a pixel
+  glVertex3f(static_cast<GLfloat>(pos.X()), static_cast<GLfloat>(pos.Y()),
+             static_cast<GLfloat>(pos.Z()));
+
+  glTexCoord2f(static_cast<GLfloat>(tex_frac_x), 0.0);
+  pos = rectDet.getRelativePosAtXY(rectDet.xpixels() - 1, 0);
+  pos += V3D(rectDet.xstep() * (+0.5), rectDet.ystep() * (-0.5),
+             0.0); // Adjust to account for the size of a pixel
+  glVertex3f(static_cast<GLfloat>(pos.X()), static_cast<GLfloat>(pos.Y()),
+             static_cast<GLfloat>(pos.Z()));
+
+  glTexCoord2f(static_cast<GLfloat>(tex_frac_x),
+               static_cast<GLfloat>(tex_frac_y));
+  pos =
+      rectDet.getRelativePosAtXY(rectDet.xpixels() - 1, rectDet.ypixels() - 1);
+  pos += V3D(rectDet.xstep() * (+0.5), rectDet.ystep() * (+0.5),
+             0.0); // Adjust to account for the size of a pixel
+  glVertex3f(static_cast<GLfloat>(pos.X()), static_cast<GLfloat>(pos.Y()),
+             static_cast<GLfloat>(pos.Z()));
+
+  glTexCoord2f(0.0, static_cast<GLfloat>(tex_frac_y));
+  pos = rectDet.getRelativePosAtXY(0, rectDet.ypixels() - 1);
+  pos += V3D(rectDet.xstep() * (-0.5), rectDet.ystep() * (+0.5),
+             0.0); // Adjust to account for the size of a pixel
+  glVertex3f(static_cast<GLfloat>(pos.X()), static_cast<GLfloat>(pos.Y()),
+             static_cast<GLfloat>(pos.Z()));
+
+  glEnd();
+  if (glGetError() > 0)
```

Comment:
`cout` will be buried on Windows. Can we get more information in this case and send it to a logger?

Reply:
@martyngigg in the follow up [PR](https://github.com/mantidproject/mantid/pull/21880) these are completely removed. There is no sense fixing it for this PR.

Reply:
Sounds good.